File: clientPython/zeroc-ice-3.7.2/java/src/IceBT/src/main/java/com/zeroc/IceBT/Instance.java
//
// Copyright (c) ZeroC, Inc. All rights reserved.
//

package com.zeroc.IceBT;

import com.zeroc.Ice.Communicator;
import com.zeroc.Ice.PluginInitializationException;

import android.bluetooth.BluetoothAdapter;

class Instance extends com.zeroc.IceInternal.ProtocolInstance
{
    Instance(Communicator communicator, short type, String protocol)
    {
        super(communicator, type, protocol, type == com.zeroc.Ice.BTSEndpointType.value);

        _communicator = communicator;

        _bluetoothAdapter = BluetoothAdapter.getDefaultAdapter();
        if(_bluetoothAdapter == null)
        {
            throw new PluginInitializationException("bluetooth adapter not available");
        }
        else if(!_bluetoothAdapter.isEnabled())
        {
            throw new PluginInitializationException("bluetooth is not enabled");
        }
    }

    void destroy()
    {
        _communicator = null;
    }

    Communicator communicator()
    {
        return _communicator;
    }

    BluetoothAdapter bluetoothAdapter()
    {
        return _bluetoothAdapter;
    }

    private Communicator _communicator;
    private BluetoothAdapter _bluetoothAdapter;
}

File: clientPython/zeroc-ice-3.7.2/csharp/src/IceSSL/Instance.cs
//
// Copyright (c) ZeroC, Inc. All rights reserved.
//

namespace IceSSL
{
    using System.Security.Authentication;
    using System.Security.Cryptography.X509Certificates;

    internal class Instance : IceInternal.ProtocolInstance
    {
        internal Instance(SSLEngine engine, short type, string protocol) :
            base(engine.communicator(), type, protocol, true)
        {
            _engine = engine;
        }

        internal SSLEngine engine()
        {
            return _engine;
        }

        internal int securityTraceLevel()
        {
            return _engine.securityTraceLevel();
        }

        internal string securityTraceCategory()
        {
            return _engine.securityTraceCategory();
        }

        internal bool initialized()
        {
            return _engine.initialized();
        }

        internal X509Certificate2Collection certs()
        {
            return _engine.certs();
        }

        internal SslProtocols protocols()
        {
            return _engine.protocols();
        }

        internal int checkCRL()
        {
            return _engine.checkCRL();
        }

        internal void traceStream(System.Net.Security.SslStream stream, string connInfo)
        {
            _engine.traceStream(stream, connInfo);
        }

        internal void verifyPeer(string address, IceSSL.ConnectionInfo info, string desc)
        {
            _engine.verifyPeer(address, info, desc);
        }

        private SSLEngine _engine;
    }
}
